build/media-suite_helper.sh

```diff
@@ -1795,3 +1829,43 @@ verify_cuda_deps() {
     fi
     enabled_any libnpp cuda-nvcc
 }
+
+check_custom_patches(){
+    local vcsFolder
+    local vcsType
+    IFS=- read vcsFolder vcsType <<< "$PWD"
```

Comment:
Looks cute. What does this do? Something like a `vcsFolder, vcsType = $PWD.split('-')`?
How does it look with a vcsFolder with hyphens?

Reply:
Something like that, this works on the assumption that the current folder is made through do_vcs and thus has ${Repo}-${git|hg|svn}

I actually don't need the vcsType part, but just in case

Reply:
I actually did not know you can assign variables in an array like that for powershell. Thanks for showing that.

Reply:
Do you know of anything better/more resilient?

Reply:
I'd use
```base
local _basedir="$(get_first_subdir)"
vcsFolder=${_basedir%-*}
vcsType=${_basedir##*-}
```, like I've used plenty through the scripts.

Reply:
I didn't even know that was valid powershell. I knew that way of tuple assignment from python and node.

Reply:
I realized that this is actually bad, mainly because it doesn't actually return the folder name. wait a bit

Reply:
Maybe I should first reverse it then split?

Reply:
Might be better for those downloaded from Sourceforge since they seem to have ${package}-${ver} generally

Reply:
I had a problem when trying to use get_first_subdir before (was only giving me .) But I'll try again once I get home)

Reply:
Yeah, non-vcs folders are a bit of a gamble. You might use the same "algorithm" that the suite uses to create the folder for those or pass it to `check_custom_patches` instead for those.